taxonomy-manager-rest-server/src/main/java/com/digirati/taxman/rest/server/infrastructure/media/writer/RdfModelMessageBodyWriter.java

```diff
@@ -0,0 +1,93 @@
+package com.digirati.taxman.rest.server.infrastructure.media.writer;
+
+import com.digirati.taxman.common.rdf.RdfModel;
+import com.digirati.taxman.common.rdf.RdfModelException;
+import com.digirati.taxman.common.rdf.RdfModelFormat;
+import com.digirati.taxman.common.rdf.annotation.jsonld.JsonLdFrame;
+import com.digirati.taxman.common.rdf.io.RdfModelWriter;
+import com.digirati.taxman.rest.MediaTypes;
+import com.github.jsonldjava.core.JsonLdOptions;
+import com.google.common.io.Resources;
+import org.apache.jena.riot.JsonLDWriteContext;
+import org.apache.jena.riot.RDFFormat;
+import org.apache.jena.sparql.util.Context;
+import org.jboss.resteasy.spi.util.FindAnnotation;
+import org.json.JSONObject;
+
+import javax.ws.rs.WebApplicationException;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.MultivaluedMap;
+import javax.ws.rs.ext.MessageBodyWriter;
+import javax.ws.rs.ext.Provider;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Type;
+import java.nio.charset.StandardCharsets;
+
+@Provider
+public class RdfModelMessageBodyWriter implements MessageBodyWriter<RdfModel> {
+
+    @Override
+    public boolean isWriteable(
+            Class<?> type, Type genericType, Annotation[] annotations, MediaType mediaType) {
+
+        return RdfModel.class.isAssignableFrom(type)
+                && (mediaType.isCompatible(MediaTypes.APPLICATION_RDF_XML)
+                        || mediaType.isCompatible(MediaTypes.APPLICATION_JSONLD_SKOS));
+    }
+
+    @Override
+    public void writeTo(
+            RdfModel model,
+            Class<?> type,
+            Type genericType,
+            Annotation[] annotations,
+            MediaType mediaType,
+            MultivaluedMap<String, Object> httpHeaders,
+            OutputStream entityStream)
+            throws IOException, WebApplicationException {
+
+        // @TODO: Handle creation and passing of `context` in a better way.
```

Comment:
Do you have any suggestions on how to resolve that?

Reply:
Personally I think we can just drop the TODO and leave it as is. We can't change the method signature to pass it in as an argument, we conditionally use a different value based on the method args, and since it goes into the bowels of Jena I don't think we can assume that the state in the context will remain unchanged to set it up as a constant. It would be nice if we could find a better way of doing this but I don't think we'll be able to find anything that works

Reply:
ACK, happy to leave this in for now. I'd thought about splitting out to `GenericRdfWriter` and `FramedJsonLdWriter`, but I don't think that's quite right either. The only purpose of the `Context` here is to create the correct configuration when we're using JSON-LD frames as output.